### PR TITLE
Arc 205 ros1 cleanup

### PR DESCRIPTION
**Pull request by john-chrosniak**
# PR Details
## Description

This PR removes ROS1 configurations that are no longer required after ROS2 migration efforts. Changes include:

1. Switching to the ROS2 version of the `carma-lightbar-driver`
2. Removing unused ROS1 launch files
3. Removing code related to `avt_vimba_camera` and `mock-camera-driver`
4. Removing unused ROS1 launch arguments
5. Removing the unused `ros2_initial_plugins` from `guidance` (No longer needed after [usdot-fhwa-stol/carma-platform#2437](https://github.com/usdot-fhwa-stol/carma-platform/pull/2437))
6. Removing the unused `traxxas_slash_4x4_platinum_2012`

## Related GitHub Issue
[usdot-fhwa-stol/carma-platform#2079](https://github.com/usdot-fhwa-stol/carma-platform/issues/2079)

## Related Jira Key

[ARC-205](https://usdot-carma.atlassian.net/browse/ARC-205)

## Motivation and Context

Removing configurations for ROS1 packages that are no longer used is needed for furthering our ROS2 migration efforts.

## How Has This Been Tested?
Tested locally using the development carma-config. Carma platform starts up and all components run without crashing.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[ARC-205]: https://usdot-carma.atlassian.net/browse/ARC-205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ail_vru_uc1_scenario/SubsystemControllerParams.yaml [hunk 1]
```diff
@@ -23,8 +23,6 @@ hardware_interface:
         - /hardware_interface/carla_gnss_driver
       ros1_required_drivers:
         - /hardware_interface/carla_driver
-      ros1_camera_drivers:
-        - /hardware_interface/carla_camera_driver
```
Comment (adev4a) [inline ail_vru_uc1_scenario/SubsystemControllerParams.yaml]: Just a note - this PR has overlapping changes with https://github.com/usdot-fhwa-stol/carma-config/pull/387/files

File: chrysler_pacifica_ehybrid_s_2019/drivers.launch [hunk 2]
```diff
@@ -45,39 +45,17 @@ If not using simulated drivers they are activated if the respective mock argumen
 
 
   <!-- Specific Drivers -->
-  <arg name="drivers" default="ssc_interface_wrapper lightbar_driver avt_vimba_camera"
+  <arg name="drivers" default="ssc_interface_wrapper lightbar_driver"
```
Comment (adev4a) [inline chrysler_pacifica_ehybrid_s_2019/drivers.launch]: Lightbar driver can be removed from here since it is included as a ros2 node in drivers.launch.py

Comment (adev4a) [reply chrysler_pacifica_ehybrid_s_2019/drivers.launch]: Also, need to double check but seems like mock_can, mock_radar, mock_camera and mock_roadway_sensor arguments don't have corresponding services in this config and can be removed

File: chrysler_pacifica_ehybrid_s_2019/drivers.launch [hunk 3]
```diff
@@ -45,39 +45,17 @@ If not using simulated drivers they are activated if the respective mock argumen
...
-    <include if="$(arg avt_vimba_camera)" ns="camera_right" file="$(find avt_vimba_camera)/launch/mono_camera.launch">
-	<arg name="guid" default=""/>
- 	<arg name="ip" default="192.168.40.10"/>
-  <arg name="camera_info_url" default="file://$(find vehicle_calibration_dir)/avt_vimba_camera/calibration_50-0536911040.yaml"/>
- 	<arg name="frame_id" default="right_optical"/>
-	<arg name="trig_timestamp_topic" default=""/>
-	<arg name="show_debug_prints" default="true"/>
-  </include>-->
-
   <!-- TODO DelphiESR Front Driver Node -->
```
Comment (adev4a) [inline chrysler_pacifica_ehybrid_s_2019/drivers.launch]: These TODO comments can be removed

File: ford_fusion_sehybrid_2019/drivers.launch [hunk 4]
```diff
@@ -43,39 +43,17 @@ If not using simulated drivers they are activated if the respective mock argumen
   <arg name="mock_lightbar"       value="$(eval 'lightbar' in arg('mock_drivers').split())"/>
 
   <!-- Specific Drivers -->
-  <arg name="drivers" default="ssc_interface_wrapper lightbar_driver avt_vimba_camera"
+  <arg name="drivers" default="ssc_interface_wrapper lightbar_driver"
```
Comment (adev4a) [inline ford_fusion_sehybrid_2019/drivers.launch]: Same comment as above regarding lightbar driver and unrequired mock_drivers

**Review by adev4a — CHANGES_REQUESTED** [review]
This looks good. There are a few additional changes that can be made here:

1. Remove drivers.launch for configs where nothing is being launched. For vehicle configs where nodes are being launched in this file, remove references to radar drivers which currently are not planned to be implemented.
2. carma_docker.launch - parameters not required in https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma/launch/carma_src.launch can be removed
3. SubsystemControllerParams: guidance_controller, ros2_initial_plugins list can be removed, that functionality was removed under https://github.com/usdot-fhwa-stol/carma-platform/pull/2437
4. docker-compose.yml: Remove references to radar drivers which are currently not planned to be implemented. The `runtime:nvidia` configuration can be moved to the platform_ros2 service since the gpu intensive nodes are in ros2
5. Let's add a reference to this github issue which is tracking all other PRs to remove unused ros1 code https://github.com/usdot-fhwa-stol/carma-platform/issues/2079

File: ford_fusion_sehybrid_2019/drivers.launch (hunk 4 above)
Comment (john-chrosniak) [reply ford_fusion_sehybrid_2019/drivers.launch]: Done

File: chrysler_pacifica_ehybrid_s_2019/drivers.launch (hunk 3 above)
Comment (john-chrosniak) [reply chrysler_pacifica_ehybrid_s_2019/drivers.launch]: Removed

File: chrysler_pacifica_ehybrid_s_2019/drivers.launch (hunk 2 above)
Comment (john-chrosniak) [reply chrysler_pacifica_ehybrid_s_2019/drivers.launch]: Lightbar driver and the unused mock drivers have been removed

**Review by adev4a — APPROVED** [review]

**Pull request merged**